diff --git a/tests/src/network_interface_test.cpp b/tests/src/network_interface_test.cpp
index ee0f21f..5190b2b 100644
--- a/tests/src/network_interface_test.cpp
+++ b/tests/src/network_interface_test.cpp
@@ -68,9 +68,5 @@ TEST_F(NetworkInterfaceTest, IterateOverInterfaces) {
     for (size_t i = 0; i < interfaces.size(); ++i) {
         // Expect this interface to be equal to itself
         EXPECT_EQ(interfaces[i], interfaces[i]);
-        // We expect to be able to construct the interface from a name 
-        // and they should still be equal
-        NetworkInterface iface(interfaces[i].name());
-        EXPECT_EQ(interfaces[i], iface);
     }
 }
