diff --git a/src/utils/radiotap_parser.cpp b/src/utils/radiotap_parser.cpp
index 8c9abb3..5a4dd78 100644
--- a/src/utils/radiotap_parser.cpp
+++ b/src/utils/radiotap_parser.cpp
@@ -65,7 +65,7 @@ const RadioTapParser::FieldMetadata RadioTapParser::RADIOTAP_METADATA[] = {
 };
 
 const uint32_t RadioTapParser::MAX_RADIOTAP_FIELD = sizeof(RADIOTAP_METADATA) /
-                                                    sizeof(FieldMetadata) + 1;
+                                                    sizeof(FieldMetadata);
 
 #if TINS_IS_LITTLE_ENDIAN
 TINS_BEGIN_PACK
@@ -135,7 +135,7 @@ void align_buffer(const uint8_t* buffer_start, const uint8_t*& buffer, uint32_t
     uint32_t offset = (buffer - buffer_start) & (n - 1);
     if (offset) {
         offset = n - offset;
-        if (TINS_UNLIKELY(offset > size)) {
+        if (TINS_UNLIKELY(buffer + offset > buffer_start + size)) {
             throw malformed_packet();
         }
         buffer += offset;
